respx/transports.py

```diff
@@ -60,6 +74,31 @@ def reset(self) -> None:
     def __getitem__(self, alias: str) -> Optional[RequestPattern]:
         return self.aliases.get(alias)
 
+    @overload
+    def pop(self, alias: str) -> RequestPattern:
+        ...
+
+    @overload
+    def pop(
+        self, alias: str, default: DefaultType
+    ) -> Union[RequestPattern, DefaultType]:
+        ...
+
+    def pop(self, alias, default=...):
```

Comment:
I wanted to retain the original behaviour of `dict.pop()`. Here are some resources on why it looks like this:
* Use of ellipsis: https://www.python.org/dev/peps/pep-0484/#arbitrary-argument-lists-and-default-argument-values
* Use of overloading: https://github.com/python/mypy/issues/3737#issuecomment-316263133
* Use of the `is` operator: https://stackoverflow.com/a/57628817/1328576